sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdConnectionStateListener.java

```diff
@@ -71,6 +77,39 @@ public RntbdConnectionStateListenerMetrics getMetrics() {
         return this.metrics;
     }
 
+    public void openConnectionIfNeeded() {
+
+        if (this.endpoint.isClosed()) {
+            return;
+        }
+
+        // do not fail here, just log
+        // this attempts to make the open connections flow
+        // best effort
+        if (this.proactiveOpenConnectionsProcessor == null) {
+            logger.warn("proactiveOpenConnectionsProcessor is null");
+            return;
+        }
+
+        // connection state listener submits an open connection task for an endpoint
+        // and only submits the task for that endpoint when the previous task has been
+        // completed
+        // it is okay to lose a task, since each task will attempt to attain a certain no. of
+        // connection as denoted by the min required no. of connections for that endpoint
+        if (this.endpointValidationInProgress.compareAndSet(false, true)) {
+            Mono.fromFuture(this.proactiveOpenConnectionsProcessor.submitOpenConnectionTaskOutsideLoop(
+                "",
+                this.endpoint.serviceEndpoint(),
+                this.addressUris.stream().findFirst().get(),
+                this.endpoint.getMinChannelsRequired()
+            ))
+            .doFinally(signalType -> {
+                this.endpointValidationInProgress.compareAndSet(true, false);
+            })
+            .subscribeOn(CosmosSchedulers.OPEN_CONNECTIONS_BOUNDED_ELASTIC)
```

Comment:
We should use `publishOn()` but worth confirming with @xinlian12

Reply:
Discussed offline.